### PR TITLE
Daniel

### PR DESCRIPTION
**Pull request by danielchen-pyc**
# Changes
- I have successfully restricted user access when the user haven't added at least one course. The breadcrumbs are initially all grayed out except the first one "Home". As the user progress through the website, the breadcrumb will be un-grayed one by one. The user can decide to go back to previous pages without graying breadcrumb again. 

## Motivation and Context
- User can jump through pages without entering any information first

## How Has This Been Tested?
- Locally tested
 
# Related Issues
- Issue #117

### CODE REVIEW

File: client/src/components/BreadCrumb.tsx [hunk 1]
```diff
@@ -7,6 +7,8 @@ import { RootState } from "../reducers/index";
 import { connect } from "react-redux";
 import { Dispatch } from "redux";
 
+var latestState = 0;
```
Comment (johnagl) [inline client/src/components/BreadCrumb.tsx:10]: We shouldn't store global variables, if we want to maintain state, we should do it in the component. Look at Courses component to see how to store state in function components (ie: useState) .

File: client/src/reducers/HomeReducer.ts [hunk 2]
```diff
@@ -29,33 +30,49 @@ const initialState: HomeReducerProps = {
...
+      // console.log(action.index, state.latestState, (action.index === 1));
+      if ((action.index > state.latestState + 1 && action.index <= 5) || (state.latestState === 1 && action.index === 2)) {
+        return state;
+      }
       return { ...state, componentIndex: action.index };
     }
     case ADDCOURSESECTIONS: {
-      return { ...state, sections: action.sections };
+      latestState = 1 <= state.latestState ? state.latestState : 2;
+      // console.log("Add course! ", latestState);
```
Comment (johnagl) [inline client/src/reducers/HomeReducer.ts:51]: I'm not a huge fan of adding all this logic into the reducer. I think we can validate a lot of this in NextPrevSwitcher component...

**Review by johnagl — COMMENTED** [review]
@ubclaunchpad/life-at-ubc Does anyone have any thoughts on implementing this? I'm thinking most of the logic should be placed in NextPrevSwitcher component instead of in the reducer - but maybe I'm just being picky.

File: client/src/reducers/HomeReducer.ts (hunk 2 above)
Comment (yehee) [reply client/src/reducers/HomeReducer.ts:51]: I second this!

**Comment by yehee** [comment]
Completely random, but if we're getting rid of breadcrumb, why don't we use something like this? https://material-ui.com/components/steppers/

**Comment by johnagl** [comment]
> Completely random, but if we're getting rid of breadcrumb, why don't we use something like this? https://material-ui.com/components/steppers/



> Completely random, but if we're getting rid of breadcrumb, why don't we use something like this? https://material-ui.com/components/steppers/

Commented in your other draft branch regarding the stepper! :)

**Comment by johnagl** [comment]
Hey @danielchen-pyc thinking about this a bit more, I think we should put this on hold. Currently the way it's implemented with NextPrevSwitcher, it's complicated to validate. @yehee has a stepper branch up which can probably be merged soon - it uses stepper from Material UI - and it avoids the complication of validating at every step.

**Comment by danielchen-pyc** [comment]
@johnagl Hello! Yes, I am working on the `stepper` branch with @yehee. I will keep you updated with what we get! Thanks!

**Comment by danielchen-pyc** [comment]
Another thing, should I close this pull request then?

**Comment by johnagl** [comment]
> Another thing, should I close this pull request then?

Yes this can be closed :).

**Pull request closed without merging**